migration_pipeline.py

```
@@ -0,0 +1,704 @@
+                    print(f"No <cassini:Legacy_Metadata> or </cassini:ISS_Specific_Attributes> found in {name}_calib.lblx")
+                    continue
+                # Find the end of the line containing </ISS_Specific_Attributes>
+                line_end = text.find('\n', iss_end)
+                if line_end == -1:
+                    line_end = len(text)
+                # Insert the block after </ISS_Specific_Attributes> with proper newline
+                updated_text = text[:line_end] + "\n" + cleaned_block + text[line_end:]
+            else:
+                # Legacy_Metadata exists in the original file, inject calibrated attributes before it
+                # Find the start of the line containing Legacy_Metadata to preserve indentation
+                line_start = text.rfind('\n', 0, insertion_point) + 1
```

Ruff (0.14.14)

[error] Local variable `line_start` is assigned to but never used

Remove assignment to unused variable `line_start`

(F841)